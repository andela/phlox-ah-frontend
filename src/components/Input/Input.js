import React from 'react';
import PropTypes from 'prop-types';
import './Input.scss';

export const Input = (props) => {
  const className = `form-input ${props.inputClassName || ''}`;
  return (
    <div className="row">
      <div className={`input-field col s${props.s}`}>
        <input
          id={props.id}
          name={props.name}
          value={props.value}
          onChange={props.onChange}
          type={props.type || 'text' }
          className={props.hasError ? `${className} input-error` : className }
          required={props.required || false}
        />
        <label
          className={`form-label ${props.labelClassName || ''}`}
          htmlFor={props.id}>
          {props.label}
        </label>
      </div>
    </div>
  );
};


Input.propTypes = {
  inputClassName: PropTypes.string,
  labelClassName: PropTypes.string,
  s: PropTypes.number,
  id: PropTypes.string,
  name: PropTypes.string,
  value: PropTypes.string,
  hasError: PropTypes.bool,
  required: PropTypes.bool,
  type: PropTypes.string,
  onChange: PropTypes.func.isRequired,
  label: PropTypes.string
};
