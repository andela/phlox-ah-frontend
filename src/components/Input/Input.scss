div.row {
  margin-bottom: 0px;
}
div.input-field input.form-input {
  margin: 0;
  padding: 1px;
  font-size: 12px;
  height: 20px;
  letter-spacing: 0.5px;
}
.input-field input[type="text"]:focus,
.input-field input[type="password"]:focus {
  box-shadow: 0 1px 0 0 #777;
  border-bottom: 1px solid #777;
  box-shadow: none;
}
div.input-field input.input-error{
  border-bottom: 1px solid red;
}
div.input-field label.form-label{
  font-size: 13px;
  height: 20px;
  top: auto;
  bottom: 26px;
  letter-spacing: o.5px;
}
.input-field > input[type]:-webkit-autofill:not(.browser-default) + label, .input-field > input[type=date]:not(.browser-default) + label, .input-field > input[type=time]:not(.browser-default) + label,
div.input-field label.form-label.active {
  -webkit-transform: translateY(0px) scale(0.8);
  transform: translateY(0px) scale(0.8);
  -webkit-transform-origin: 0 0;
  transform-origin: 0 0;
  font-size: 13px;
}
