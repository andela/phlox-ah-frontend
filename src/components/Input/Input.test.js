
import React from 'react';
import { shallow } from 'enzyme';
import configureMockStore from 'redux-mock-store';

import { Input } from './Input';

const props = {
  onChange: jest.fn(),
  className: 'class-name',
  type: 'text',
  name: 'input',
  id: 'id',
  value: 'value',
  label: 'label',
  required: true,
  hasError: true,
  labelClassName: 'labelClassName'
};

let component;


describe('<Input/>', () => {
  beforeEach(() => {
    component = shallow(
        <Input {...props} />
    );
  });
  it('should render without throwing an error', () => {
    expect(component).toMatchSnapshot();
  });
  it('should have a div tag with class row', () => {
    expect(component.find('div.row').exists()).toBe(true);
  });
  it('should have a div.input-field tag', () => {
    expect(component.find('div.input-field').exists()).toBe(true);
  });
  it('should have a input tag', () => {
    expect(component.find('input').exists()).toBe(true);
  });
  it('should have a label tag', () => {
    expect(component.find('label').exists()).toBe(true);
  });
});
