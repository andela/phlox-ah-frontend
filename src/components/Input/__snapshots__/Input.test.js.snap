// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input/> should render without throwing an error 1`] = `
<div
  className="row"
>
  <div
    className="input-field col sundefined"
  >
    <input
      className="form-input  input-error"
      id="id"
      name="input"
      onChange={[MockFunction]}
      required={true}
      type="text"
      value="value"
    />
    <label
      className="form-label labelClassName"
      htmlFor="id"
    >
      label
    </label>
  </div>
</div>
`;
